test/python/test_dagcircuit.py

```diff
@@ -264,6 +264,16 @@ def test_dag_ops_on_wire(self):
         with self.assertRaises(DAGCircuitError):
             next(self.dag.nodes_on_wire((reg, 7)))
 
+    def test_remove_op_node(self):
+        """ Test remove_op_node method."""
+        self.dag.apply_operation_back(HGate(self.qubit0))
+
+        op_nodes = self.dag.gate_nodes()
+        h_gate = op_nodes.pop()
+        self.dag.remove_op_node(h_gate)
+
+        self.assertEqual(len(self.dag.gate_nodes()), 0)
+
```

Comment:
Can you add a test for the case where we're removing a node from a larger graph? At least that we're removing the right node and that the remaining graph structure is correct (mainly that the re-wired edges are consistent).

Reply:
fixed in af2d8743